### DISCUSSION TITLE
Granting user role to sysadmin or viceversa

### DISCUSSION THREAD

**Discussion by rajm1230** [littleK0i/SnowDDL#138] [General]
How do I grant automatically created  user / business role by Snowddl to sysadmin or built in snowflake role  using yaml when creating business or user  role 

tried to use global roles but it just grant snowflake role to business role that being created by snowddl 

please provide valid yaml example for the same

**Comment by littleK0i** [comment]
Everything created by SnowDDL is directly or implicitly owned by or granted to SnowDDL admin role (`SNOWDDL_ADMIN` in documentation): https://docs.snowddl.com/guides/other-guides/admin

With first setup option `SNOWDDL_ADMIN` sits in the middle between `SYSADMIN + SECURITYADMIN` and `ACCOUNTADMIN`.

Full hierarchy looks like this:
```
<roles_created_by_snowddl>
SYSADMIN                              ->     SNOWDDL_ADMIN       ->           ACCOUNTADMIN
SECURITYADMIN
``` 

ACCOUNTADMIN's should be able to see and access everything.

I suspect, roles created by SnowDDL may not be granted to `SYSADMIN` or `SECURITYADMIN`, since it would create a closed loop, and Snowflake does not permit loops. But it should not be a problem in general. You should still be able to do everything.

This structure it used intentionally to separate objects managed by SnowDDL from objects managed by other tools. It helps to prevent accidental damage.

**Comment by littleK0i** [comment]
Could you provide a bit more context, why this grant is important?

We might be able to figure out a better way to achieve the same result.
